### PR TITLE
Added pkg/spire mocking and testing

### PR DESCRIPTION
**Pull request by lumjjb**

### CODE REVIEW

File: pkg/spire/spire_mock.go [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+
+	// SignOverride contains the function to overwrite a call to Sign
+	SignOverride func(ctx context.Context, results []v1beta1.PipelineResourceResult) ([]v1beta1.PipelineResourceResult, error)
+}
+
+const (
+	controllerSvid = "CONTROLLER_SVID_DATA"
+)
+
+func (_ *SpireMockClient) mockSign(content, signedBy string) string {
```
Comment (pxp928) [inline pkg/spire/spire_mock.go:71]: is `_ *SpireMockClient` needed if its not being used? We can make a standalone function?

File: pkg/spire/spire_mock.go [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+
+func (sc *SpireMockClient) mockVerify(content, sig, signedBy string) bool {
+	return sig == sc.mockSign(content, signedBy)
+}
+
+func (_ *SpireMockClient) GetIdentity(tr *v1beta1.TaskRun) string {
+	return fmt.Sprintf("/ns/%v/taskrun/%v", tr.Namespace, tr.Name)
+}
+
+func (sc *SpireMockClient) AppendStatusInternalAnnotation(ctx context.Context, tr *v1beta1.TaskRun) error {
```
Comment (pxp928) [inline pkg/spire/spire_mock.go:83]: missing a check for `AppendStatusInternalAnnotationOverride`

File: pkg/spire/spire_mock_test.go [hunk 3]
```diff
@@ -0,0 +1,710 @@
...
+	spireMockClient := &SpireMockClient{}
+	var (
+		cc SpireControllerApiClient = spireMockClient
+	)
+
+	ctx := context.Background()
+	trs := testTaskRuns()
+	tr1, tr2 := trs[0], trs[1]
+
+	trs = testTaskRuns()
```
Comment (pxp928) [inline pkg/spire/spire_mock_test.go:92]: calling `trs = testTaskRuns()` twice

**Review by pxp928 — APPROVED** [review]
LGTM! Overall very comprehensive checks. The mocked spire does cover all public functions.

File: pkg/spire/spire_mock.go (hunk 2 above)
Comment (lumjjb) [reply pkg/spire/spire_mock.go:83]: fixed

File: pkg/spire/spire_mock.go (hunk 1 above)
Comment (lumjjb) [reply pkg/spire/spire_mock.go:71]: I wanted to make sure we didn't accidentally call it from one of our packages, or to get misused.

**Comment by lumjjb** [comment]
Ready for another round of review!

File: pkg/pod/status_test.go [hunk 4]
```diff
@@ -34,6 +38,353 @@ import (
 
 var ignoreVolatileTime = cmp.Comparer(func(_, _ apis.VolatileTime) bool { return true })
 
+func TestMakeTaskRunStatusVerify(t *testing.T) {
+	sc := &spire.SpireMockClient{}
+	processConditions := cmp.Transformer("sortConditionsAndFilterMessages", func(in []apis.Condition) []apis.Condition {
```
Comment (pxp928) [inline pkg/pod/status_test.go:43]: Question - why are these `tansformers` needed?

**Review by pxp928 — APPROVED** [review]
LGTM! Great overall unit test coverage. Just had the one question

**Review by pxp928 — APPROVED** [review]

**Pull request merged**